### PR TITLE
Proxy discovery client.

### PR DESCRIPTION
**Pull request by evanlinjin**

### CODE REVIEW

File: pkg/proxydisc/client.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+func (c *HTTPClient) addr(path string) string {
+	return c.conf.DiscAddr + path
+}
+
+// Auth returns the internal httpauth.Client
+func (c *HTTPClient) Auth(ctx context.Context) (*httpauth.Client, error) {
+	if c.auth != nil {
+		return c.auth, nil
+	}
+	fmt.Println(c.conf.DiscAddr)
```
Comment (Darkren) [inline pkg/proxydisc/client.go]: Is it intended ot was left after debugging?

**Review by Darkren — COMMENTED** [review]
Great job!

File: pkg/proxydisc/client.go (hunk 1 above)
Comment (evanlinjin) [reply pkg/proxydisc/client.go]: Left after debugging. Thanks for the heads up.

File: pkg/proxydisc/error.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+// HTTPError represents an HTTP error.
+type HTTPError struct {
+	Status int    `json:"http_status"` // HTTP Status.
```
Comment (nkryuchkov) [inline pkg/proxydisc/error.go]: Is there a reason why struct and JSON names differ?

**Review by nkryuchkov — APPROVED** [review]
The code looks fine, great job!

File: pkg/proxydisc/error.go (hunk 2 above)
Comment (evanlinjin) [reply pkg/proxydisc/error.go]: No reason. I'll change it.

**Pull request merged**